src/operations/indexes.ts

```diff
@@ -196,7 +219,25 @@ export class CreateIndexesOperation<
     this.options = options ?? {};
     this.collectionName = collectionName;
 
-    this.indexes = indexes;
+    // Ensure we generate the correct name if the parameter is not set
```

Comment:
Can we move this code into a function?  We discussed this a bit in the office, but I still think for readability it should be broken out.  The fact that a comment is necessary to explain what this block of code is doing strongly indicates that this should be broken out into a function.

Reply:
Sure done!